selfdrive/car/fw_versions.py

```diff
@@ -284,18 +285,28 @@ def match_fw_to_car_exact(fw_versions_dict):
 
 
 def match_fw_to_car(fw_versions, allow_fuzzy=True):
-  fw_versions_dict = build_fw_dict(fw_versions)
-  matches = match_fw_to_car_exact(fw_versions_dict)
+  versions = get_interface_attr('FW_VERSIONS', ignore_none=True)
+
+  # Try exact matching first
+  exact_matches = [True]
+  if allow_fuzzy:
+    exact_matches.append(False)
 
-  exact_match = True
-  if allow_fuzzy and len(matches) == 0:
-    matches = match_fw_to_car_fuzzy(fw_versions_dict)
+  for exact_match in exact_matches:
+    # For each brand, attempt to fingerprint using FW returned from its queries
+    for brand in versions.keys():
+      fw_versions_dict = build_fw_dict(fw_versions, filter_brand=brand)
 
-    # Fuzzy match found
-    if len(matches) == 1:
-      exact_match = False
+      if exact_match:
+        matches = match_fw_to_car_exact(fw_versions_dict)
+      else:
+        matches = match_fw_to_car_fuzzy(fw_versions_dict)
 
-  return exact_match, matches
+      if len(matches) == 1:
+        return exact_match, matches
+
+  else:
+    return True, []
```

Comment:
All the places that use this function (car_helpers, test_fw_fingerprint), they only use the return values if the length of matches is exactly 1. So checking `== 1` above, and returning empty list here changes no behavior.